File: ReflectDemo/app/src/main/java/hu/aut/reflectdemo/MainActivity.java
package hu.aut.reflectdemo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import java.lang.reflect.Field;

public class MainActivity extends AppCompatActivity {

    private TextView tvData;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvData = (TextView) findViewById(R.id.tvData);

        showReflect();
    }

    private void showReflect() {
        Class classToUse  = null;
        try {
            classToUse = Class.forName("android.app.NotificationManager");
            Field[] classFields = classToUse.getDeclaredFields();
            tvData.setText("");
            for(Field f : classFields){
                tvData.append(f.getName()+"\n");
            }
        } catch (ClassNotFoundException e) {
            e.printStackTrace();
        }


    }
}

File: ReflectionDemo/app/src/main/java/hu/bme/aut/reflectiondemo/MainActivity.kt
package hu.bme.aut.reflectiondemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import java.lang.reflect.Field
import java.lang.reflect.Method


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        showReflect()
    }

    private fun showReflect() {
        var classToUse: Class<*>? = null
        try {
            classToUse = Class.forName("android.app.NotificationManager")
            val classMethods: Array<Method> = classToUse.declaredMethods
            tvData.text = ""
            for (f in classMethods) {
                tvData!!.append(f.getName().toString() + "\n")
            }
        } catch (e: ClassNotFoundException) {
            e.printStackTrace()
        }
    }
}

